test.js

```diff
@@ -1,147 +1,106 @@
-      taskText.textContent = task.text;
-      taskText.className = 'task-text';
-      taskElement.appendChild(taskText);
-
-      // Edit Button
-      const editButton = document.createElement('button');
-      editButton.textContent = 'Edit';
-      editButton.className = 'edit-btn';
-      editButton.onclick = () => editTask(task.id);
-      taskElement.appendChild(editButton);
-
-      // Delete Button (Forgot to add a delete button handler)
-      const deleteButton = document.createElement('button');
-      deleteButton.textContent = 'Delete';
-      deleteButton.className = 'delete-btn';
-      taskElement.appendChild(deleteButton);
-
-      // Mark as Completed Button
-      const completeButton = document.createElement('button');
-      completeButton.textContent = task.completed ? 'Undo' : 'Complete';
-      completeButton.className = 'complete-btn';
-      completeButton.onclick = () => toggleComplete(task.id);
-      taskElement.appendChild(completeButton);
-
-      taskList.appendChild(taskElement);
-    });
-  }
-}
-
-// Function to add a new task
-function addTask(text) {
-  if (!text.trim()) {
-    showError('Task cannot be empty.');
+// A non-optimized to-do list application with redundant logic and poor performance
+let toDoList = [];
+// Function to add a task (overcomplicated and inefficient)
+function addTask(task) {
+  if (task === "") {
+    alert("Task cannot be empty!");
     return;
   }
-
-  const newTask = {
-    id: Date.now(), // Using timestamp as a unique ID
-    text: text,
-    completed: false,
+  let taskId = Date.now(); // Using Date.now() as a unique ID, not optimal for large data
+  let taskObject = {
+    id: taskId,
+    text: task,
+    completed: false
   };
-
-  tasks.push(newTask);
-  renderTasks();
-  taskInput.value = ''; // Clear the input field
-  hideError();
-}
-
-// Function to edit an existing task
-function editTask(id) {
-  const task = tasks.find(t => t.id === id);
-  if (!task) return;
-
-  const newText = prompt('Edit task:', task.text);
-  if (newText !== null && newText.trim()) {
-    task.text = newText;
-    renderTasks();
+  toDoList.push(taskObject);
+  // Rebuild the entire list every time
+  let listContainer = document.getElementById('task-list');
+  listContainer.innerHTML = ''; // Inefficient clearing of the entire list
+  for (let i = 0; i < toDoList.length; i++) {
+    let taskItem = document.createElement('li');
+    taskItem.textContent = toDoList[i].text;
+    listContainer.appendChild(taskItem);
   }
 }
-
-// Function to delete a task (No implementation, error here!)
-function deleteTask(id) {
-  tasks = tasks.filter(t => t.id !== id);
-  renderTasks();
-}
-
-// Function to toggle the completion status of a task
-function toggleComplete(id) {
-  const task = tasks.find(t => t.id === id);
-  if (!task) return;
-
-  task.completed = !task.completed;
-  renderTasks();
+// Function to mark a task as completed (no proper search)
```

Comment:
Line 25: Inefficient DOM manipulation detected. Instead of updating the DOM in a loop, consider using a DocumentFragment to batch updates and improve performance.